xwiki-platform-core/xwiki-platform-web/src/main/webapp/templates/register_macros.vm

```diff
@@ -0,0 +1,219 @@
+## Defines what server generated error messages should look like
+## The error message when a field is entered incorrectly
+#set ($failureMessageParams = {'class': 'LV_validation_message LV_invalid'})
+## 'LV_validation_message LV_invalid' depends on this:
+$xwiki.get('ssfx').use('uicomponents/widgets/validation/livevalidation.css', true)
+##
+## The * next to the fields to denote they are mandatory.
+#set ($fieldMandatoryStar = {'class': 'xRequired'})
+##
+#macro (definePasswordFields, $fields, $passwordFieldName, $confirmPasswordFieldName)
+  #set ($field =
+    {'name': $passwordFieldName,
+      'label': $services.localization.render('core.register.password'),
+      'params': {
+        'type': 'password',
+        'autocomplete': 'off',
+        'size': '60'
+      },
+      'validate': {
+        'mandatory': {
+          'failureMessage': $services.localization.render('core.validation.required.message')
+        },
+        'regex': {
+          'pattern': '/.{6,}/',
+          'failureMessage': $services.localization.render('xe.admin.registration.passwordTooShort')
+        }
+      }
+    })
+  #set ($discard = $fields.add($field))
+##
+## The confirm password field, mandatory, must match password field, and must also be 6+ characters long.
+  #set ($field =
+    {'name': $confirmPasswordFieldName,
+      'label': $services.localization.render('core.register.passwordRepeat'),
+      'params': {
+        'type': 'password',
+        'autocomplete': 'off',
+        'size': '60'
+      },
+      'validate': {               
+        'mandatory': {
+          'failureMessage': $services.localization.render('core.validation.required.message')
+        },
+        'mustMatch': {                
+          'name': $passwordFieldName,
+          'failureMessage': $services.localization.render('xe.admin.registration.passwordMismatch')
+        },                
+        'regex': {
+          'pattern': '/.{6,}/',
+          'failureMessage': $services.localization.render('xe.admin.registration.passwordTooShort')
+        }
+      }
+    })
+  #set ($discard = $fields.add($field))
+#end
+##
+#*
+ * Generate HTML form.
+ *
+ * @param $fields The array of fields to use for generating HTML code.
+ * @param $request The request that is made by submitting the form.
+ *#
+#macro (generateHtml, $fields, $request)
+  ## Put the same values back into the fields (if is there any problem with a field from the request that is made).
+  #getParams($fields, $request)
+  <dl>
+  #foreach ($field in $fields)
+    #if ($field.get('name'))
```

Comment:
In velocity, you can simplify all those `.get('something')` into `.something` directly and the code should look much more readable.